exec/java-exec/src/test/java/org/apache/drill/exec/record/TestMaterializedField.java

```diff
@@ -84,4 +89,22 @@ public void testClone() {
 
   }
 
+  @Test // DRILL-4264
+  public void testSchemaPathToMaterializedFieldConverting() {
```

Comment:
No test here for names with dots?

Also, still not convinced of the semantics: not sure we want to create a tree of maps given a nested path. Instead, we need a way to refer to an existing tree and either navigate down that tree to find existing maps, or create a map as needed when a node is missing.

Note that much work has been done in this area in PR #914.